From 55bf73fe6995f70adb9a50a8cb7cdea5c9aa8b41 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Tue, 25 Jul 2017 16:09:24 +0800
Subject: [PATCH] fix build issue for kernel 4.4

src/mac80211.c:1576:13: error: assignment to expression with array type
  hw->flags  = 0;
             ^
src/mac80211.c:1577:12: error: assignment to expression with array type
  hw->flags |= IEEE80211_HW_SW_CRYPTO_CONTROL;

Signed-off-by: Yangbo Lu <yangbo.lu>
---
 src/mac80211.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/mac80211.c b/src/mac80211.c
index f838dbf..f4ee645 100644
--- a/src/mac80211.c
+++ b/src/mac80211.c
@@ -1544,6 +1544,7 @@ const struct ieee80211_ops fsl_wifi_hw_ops = {
 	.sta_statistics = fsl_wifi_mac_sta_statistics,
 };
 
+#define ieee80211_hw_set(hw, flg)	_ieee80211_hw_set(hw, IEEE80211_HW_##flg)
 
 void fsl_wifi_mac_setup(struct fsl_wifi_dev* fsl_wifi_dev)
 {
@@ -1573,8 +1574,7 @@ void fsl_wifi_mac_setup(struct fsl_wifi_dev* fsl_wifi_dev)
 	// 	    IEEE80211_HW_SUPPORTS_CLONED_SKBS|
 	// 	    IEEE80211_HW_SW_CRYPTO_CONTROL;
 	//hw->flags = IEEE80211_SINGLE_HW_SCAN_ON_ALL_BANDS;
-	hw->flags  = 0;
-	hw->flags |= IEEE80211_HW_SW_CRYPTO_CONTROL;
+	ieee80211_hw_set(hw, SW_CRYPTO_CONTROL);
 	dev_info(fsl_wifi_dev->dev, "registering ieee80211 hw...\n");
 	hw->wiphy->cipher_suites = fsl_cipher_suites;
 	hw->wiphy->n_cipher_suites = ARRAY_SIZE(fsl_cipher_suites);
-- 
2.1.0.27.g96db324

